### PR TITLE
NO-ISSUE: Add Capability and FeatureGate checks to OLMv1 tests

### PR DESCRIPTION
**Pull request by tmshort**
This required:

```
go get github.com/openshift/api@a2817b89f7e0989016967055a142d5e88bed18de
go get github.com/openshift/client-go@923091dd2b1a3d6fe75a06cc4f45972a3ddf84c2
go get github.com/openshift/library-go@0064ad7bd060b9fd52f7840972c1d3e72186d0f0
// update the replaces of k8s.io/kubernetes to github.com/openshift/kubernetes@v1.30.1-0.20241111135607a3f31a4f2c58 
go mod tidy
go mod vendor
```
/hold until capability is fully implemented

### CODE REVIEW

File: test/extended/olm/olmv1.go [hunk 1]
```diff
@@ -163,3 +159,14 @@ var _ = g.Describe("[sig-olmv1] OLMv1 operator installation", func() {
 		o.Expect(err).NotTo(o.HaveOccurred())
 	})
 })
+
+func checkTestSkip(ctx context.Context, oc *exutil.CLI) {
+	if !exutil.IsTechPreviewNoUpgrade(ctx, oc.AdminConfigClient()) {
```
Comment (wking) [inline test/extended/olm/olmv1.go:164]: is there a way for tests to check the [`NewOLM` feature gate][1], so you have one less thing you need to bump when that is added to the default feature set?  This line hard-codes the current state, where [the `NewOLM` feature gate is in tech-preview, but not yet in the default feature set][2].

[1]: https://github.com/openshift/api/pull/2061/files#diff-fdab2a007eec159636f78603e2daf39204052afecb564d7ccbf715f881106c44R18
[2]: https://github.com/openshift/api/blob/a2817b89f7e0989016967055a142d5e88bed18de/features/features.go#L493-L498

Comment (tmshort) [reply test/extended/olm/olmv1.go:164]: Done, I think...

Comment (tmshort) [reply test/extended/olm/olmv1.go:164]: I didn't get rid of the TechPreview check yet, because I wasn't sure of the status of the FeatureGate.

**Comment by tmshort** [comment]
Closing in favor of #29290

**Pull request closed without merging**